patches/server/0726-Added-PlayerAngerPiglinByInteractionEvent.patch

```diff
@@ -0,0 +1,133 @@
+index d6a3f3a2edae806b0ebf5bf5ac445116c0d64535..a36633bd7a6e13e26305cfd5ab7090070ba0bd0f 100644
+--- a/src/main/java/net/minecraft/world/level/block/Block.java
++++ b/src/main/java/net/minecraft/world/level/block/Block.java
+@@ -470,7 +470,7 @@ public class Block extends BlockBehaviour implements ItemLike {
+     public void playerWillDestroy(Level world, BlockPos pos, BlockState state, Player player) {
+         this.spawnDestroyParticles(world, player, pos, state);
+         if (state.is((Tag) BlockTags.GUARDED_BY_PIGLINS)) {
+-            PiglinAi.angerNearbyPiglins(player, false);
++            PiglinAi.angerNearbyPiglins(player, false, net.minecraft.server.MCUtil.toLocation(world, pos), null); // Paper
+         }
+ 
+         world.gameEvent((Entity) player, GameEvent.BLOCK_DESTROY, pos);
+diff --git a/src/main/java/net/minecraft/world/level/block/ChestBlock.java b/src/main/java/net/minecraft/world/level/block/ChestBlock.java
+index d980a556785b52fe827310b83638139df0816b11..312675b29e2abce2b3392572f91c8f49ed9ed5c4 100644
+--- a/src/main/java/net/minecraft/world/level/block/ChestBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/ChestBlock.java
+@@ -269,7 +269,8 @@ public class ChestBlock extends AbstractChestBlock<ChestBlockEntity> implements
+             if (itileinventory != null) {
+                 player.openMenu(itileinventory);
+                 player.awardStat(this.getOpenChestStat());
+-                PiglinAi.angerNearbyPiglins(player, true);
++                org.bukkit.block.Block block = org.bukkit.craftbukkit.block.CraftBlock.at(world, pos); // Paper
++                PiglinAi.angerNearbyPiglins(player, true, block.getLocation(), block.getState() instanceof org.bukkit.inventory.InventoryHolder ? (org.bukkit.inventory.InventoryHolder) block.getState() : null); // Paper
+             }
+ 
+             return InteractionResult.CONSUME;
+diff --git a/src/main/java/net/minecraft/world/level/block/EnderChestBlock.java b/src/main/java/net/minecraft/world/level/block/EnderChestBlock.java
+index 7e45c97acce83a9fe8ada486e9fcdafe58769736..13de2e22adca30663c2bf34ded626776468b5291 100644
+--- a/src/main/java/net/minecraft/world/level/block/EnderChestBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/EnderChestBlock.java
+@@ -89,7 +89,8 @@ public class EnderChestBlock extends AbstractChestBlock<EnderChestBlockEntity> i
+                     return ChestMenu.threeRows(syncId, inventory, playerEnderChestContainer);
+                 }, CONTAINER_TITLE));
+                 player.awardStat(Stats.OPEN_ENDERCHEST);
+-                PiglinAi.angerNearbyPiglins(player, true);
++                org.bukkit.block.Block block = org.bukkit.craftbukkit.block.CraftBlock.at(world, pos); // Paper
++                PiglinAi.angerNearbyPiglins(player, true, block.getLocation(), block.getState() instanceof org.bukkit.inventory.InventoryHolder ? (org.bukkit.inventory.InventoryHolder) block.getState() : null); // Paper
```

Comment:
iirc Ender Chests are not inventory holders

Reply:
yeah, they aren't I prob just copy pasted what I had for other "chests"